### PR TITLE
Vb/feat/22/move model generation to sdk

### PR DESCRIPTION
**Pull request by vadimbasko**
contains test-rework #17

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,27 @@ sudo: false
...
+    - name: "Test e2e:polymer2"
+      script: npm run test:e2e:polymer2
+
+    - name: "Test e2e:polymer2-ts"
+      script: npm run test:e2e:polymer2-ts
+
+    - name: "Test e2e:react:mpg"
+      script: npm run test:e2e:react:mpg
+
+    - name: "Test e2e:react:trusty-dragon"
```
Comment (web-devel) [inline .travis.yml]: What does `trusty-dragon` mean ?

File: package.json [hunk 2]
```diff
@@ -31,17 +38,34 @@
...
+    "extends": "@istanbuljs/nyc-config-typescript",
+    "all": true,
+    "include": [
+      "src/**/*.ts"
+    ],
+    "exclude": [
+      "src/test/**",
+      "src/generators/polymer2/**",
+      "src/generators/polymer2-typescript/**",
+      "src/generators/app/template/**"
```
Comment (web-devel) [inline package.json:64]: Strange path, also react seems to be missing

File: src/generators/react-typescript/entity-management/template/EntityManagementBrowser.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import * as React from "react";
 import {observer} from "mobx-react";
-import {Card, List, Icon, Modal} from "antd";
+import {Card, Icon, Modal} from "antd";
```
Comment (web-devel) [inline src/generators/react-typescript/entity-management/template/EntityManagementBrowser.tsx]: We should not delete `List` component import as it's being used in template

File: src/generators/sdk/README.md [hunk 4]
```diff
@@ -6,6 +6,11 @@ rest [services](https://doc.cuba-platform.com/manual-latest/rest_api_v2_services
 and [queries](https://doc.cuba-platform.com/manual-latest/rest_api_v2_queries_config.html) 
 as Typescript classes.
 
+SDK could be generated in different configurations depends on option passed to ```gen-cuba-front``` command.
```
Comment (web-devel) [inline src/generators/sdk/README.md]: I suggest:
> It's possible to generate the following configurations of sdk depending on your needs:

File: src/generators/sdk/model/model-utils.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+export function collectAttributesFromHierarchy(entity: Entity, projectModel: ProjectModel): EntityAttribute[] {
+  let attrs: EntityAttribute[] = entity.attributes;
+
+  const allEntities: Partial<Entity>[] = ([] as Partial<Entity>[])
+    .concat(projectModel.entities)
+    .concat(projectModel.baseProjectEntities ? projectModel.baseProjectEntities : []);
+
+  let {parentClassName, parentPackage} = entity;
+
+  while (parentClassName && parentPackage && parentClassName.length > 0 && parentPackage.length > 0) {
```
Comment (web-devel) [inline src/generators/sdk/model/model-utils.ts]: looks weird )

**Review by web-devel — CHANGES_REQUESTED** [review]

File: src/generators/react-typescript/entity-management/template/EntityManagementBrowser.tsx (hunk 3 above)
Comment (vadimbasko) [reply src/generators/react-typescript/entity-management/template/EntityManagementBrowser.tsx]: fixed, conditional import and test will be impleented in #26

File: src/generators/sdk/README.md (hunk 4 above)
Comment (vadimbasko) [reply src/generators/sdk/README.md]: text changed

File: package.json (hunk 2 above)
Comment (vadimbasko) [reply package.json:64]: polymer app excluded from coverage - it's ok for now

File: src/generators/sdk/model/model-utils.ts (hunk 5 above)
Comment (vadimbasko) [reply src/generators/sdk/model/model-utils.ts]: fixed

File: .travis.yml (hunk 1 above)
Comment (vadimbasko) [reply .travis.yml]: renamed with new-project

**Pull request merged**